pkg/collector/parser/receiver_jaeger.go

```diff
@@ -87,6 +92,9 @@ func (j *JaegerReceiverParser) Ports() ([]corev1.ServicePort, error) {
 				}
 			}
 
+			// set the appropriate TCP/UDP protocol for this kind of receiver protocol
+			protocolPort.Protocol = protocol.protocol
```

Comment:
I'm slightly annoyed that we have the same term (protocol) with two meanings here as part of the same struct: the data protocol (gRPC/protobuf, Thrift-compact, Thrift-binary, ...) and the transport protocol (TCP/UDP). Your comment above saves this ambiguity from confusion. Given that I don't have a better proposal, I think it's OK to leave it as is, but if you have another idea, I'm open to consider it.

Reply:
I agree it's confusing, and preferably i'd like to make it unambiguous by default to avoid the need for a clarifying comment. Referring to it as the transport protocol in the struct would be an improvement, although `protocolPort.Protocol` will remain somewhat ambiguous regardless. It's still an overall improvement though, so i'll get another commit prepared.